libs/core/src/main/java/org/opensearch/core/compress/CompressorRegistry.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.core.compress;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.annotation.InternalApi;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.compress.spi.CompressorProvider;
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+/**
+ * A registry that wraps a static Map singleton which holds a mapping of unique String names (typically the
+ * compressor header as a string) to registerd {@link Compressor} implementations.
+ *
+ * This enables plugins, modules, extensions to register their own compression implementations through SPI
+ *
+ * @opensearch.experimental
+ * @opensearch.internal
+ */
+@InternalApi
+public final class CompressorRegistry {
+    /** No compression singleton - we still register so users can specify NONE in the API*/
+    public static final Compressor NONE;
```

Comment:
Seems like this one could be gone, we have `CompressorRegistry::none()` to do exactly that

Reply:
That's true. I kept it to avoid a Map lookup but it's not critical path code so I'm happy to remove it.